cmd/export_dashboards.go

```diff
@@ -88,12 +110,171 @@ func exportDashboardsCmd(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func promptDashboardIDs(ctx context.Context, kibanaClient *kibana.Client) ([]string, error) {
+type selectDashboardOptions struct {
+	ctx            context.Context
+	kibanaClient   *kibana.Client
+	kibanaVersion  kibana.VersionInfo
+	defaultPackage string
+}
+
+// selectDashboardIDs prompts the user to select dashboards to export. It handles
+// different flows depending on whether the Kibana instance is a Serverless environment or not.
+// In non-Serverless environments, it prompts directly for dashboard selection.
+// In Serverless environments, it first prompts to select an installed package or choose
+// to export new dashboards, and then prompts for dashboard selection accordingly.
+func selectDashboardIDs(options selectDashboardOptions) ([]string, error) {
+	if options.kibanaVersion.BuildFlavor != kibana.ServerlessFlavor {
+		// This method uses a deprecated API to search for saved objects.
+		// And this API is not available in Serverless environments.
+		dashboardIDs, err := promptDashboardIDsNonServerless(options.ctx, options.kibanaClient)
```

Comment:
When/if this deprecated API is removed we can remove this if and use the same approach as in serverless. Or do you think it would be worth to do this change now?